svgpathtools/svg2paths.py

```diff
@@ -144,30 +174,139 @@ def dom2dict(element):
         values = [val.value for val in list(element.attributes.values())]
         return dict(list(zip(keys, values)))
 
-    # Use minidom to extract path strings from input SVG
-    paths = [dom2dict(el) for el in doc.getElementsByTagName('path')]
-    d_strings = [el['d'] for el in paths]
-    attribute_dictionary_list = paths
-
-    # Use minidom to extract polyline strings from input SVG, convert to
-    # path strings, add to list
-    if convert_polylines_to_paths:
-        plins = [dom2dict(el) for el in doc.getElementsByTagName('polyline')]
-        d_strings += [polyline2pathd(pl['points']) for pl in plins]
-        attribute_dictionary_list += plins
-
-    # Use minidom to extract polygon strings from input SVG, convert to
-    # path strings, add to list
-    if convert_polygons_to_paths:
-        pgons = [dom2dict(el) for el in doc.getElementsByTagName('polygon')]
-        d_strings += [polygon2pathd(pg['points']) for pg in pgons]
-        attribute_dictionary_list += pgons
-
-    if convert_lines_to_paths:
-        lines = [dom2dict(el) for el in doc.getElementsByTagName('line')]
-        d_strings += [('M' + l['x1'] + ' ' + l['y1'] +
-                       'L' + l['x2'] + ' ' + l['y2']) for l in lines]
-        attribute_dictionary_list += lines
+    def parse_trafo(trafo_str):
+        """Returns six matrix elements for a matrix transformation for any 
+        valid SVG transformation string."""
+        trafos = trafo_str.split(')')[:-1]
+        trafo_matrix = np.array([1., 0., 0., 0., 1., 0., 0., 0., 1.]).reshape(
+            (3, 3))  # Start with neutral matrix
+
+        for trafo_sub_str in trafos:
+            trafo_sub_str = trafo_sub_str.lstrip(', ')
+            value_str = trafo_sub_str.split('(')[1]
+            values = list(map(float, value_str.split(',')))
+            if 'translate' in trafo_sub_str:
+                x = values[0]
+                y = values[1] if (len(values) > 1) else 0.
+                trafo_matrix = np.dot(trafo_matrix, np.array(
+                    [1., 0., x, 0., 1., y, 0., 0., 1.]).reshape((3, 3)))
+            elif 'scale' in trafo_sub_str:
+                x = values[0]
+                y = values[1] if (len(values) > 1) else 0.
+                trafo_matrix = np.dot(trafo_matrix,
+                                      np.array([x, 0., 0., 0., y, 0., 0., 0.,
+                                                1.]).reshape((3, 3)))
+            elif 'rotate' in trafo_sub_str:
+                a = values[0] * np.pi / 180.
+                x = values[1] if (len(values) > 1) else 0.
+                y = values[2] if (len(values) > 2) else 0.
```

Comment:
For `rotate`, according to the spec, either x and y should **both** be specified, or **neither** of them should be specified. If x is specified without y, then the SVG is malformed.